### PR TITLE
feat: Add testTags for e2e tests

### PR DESCRIPTION
**Pull request by deepanchal**
## Description

Adding testTags to every single kotlin file would be added over multiple PRs to keep PRs smaller and reviewable.

- feat(cases): add testTags on `CaseAction` enum
- feat(cases): set testTag dynamically on CaseActionButton & CaseActionFlatButton using testTag from CaseAction enum
- feat(cases): set testTags for GoogleMap and all overlay CrisisCleanupFabs in CasesScreen
- feat(designsystem): add testTag on CrisisCleanupNavigationBar, dynamically infer testTag from label in CrisisCleanupNavigationBarItem
- feat(authentication): add testTags on ui elements on login & profile screen in AuthenticateScreen
- style: format maestro test files
- feat(cases): set testTag for My location fab button
- feat(cases): add testTag for back btn on cases filter screen
- feat(designsystem): set dynamic testTag on `WorkTypePrimaryAction`
- feat(cases): add testTags on `CasesScreenTableView`

## Screenshots

N/A

## Related Issue

Refs: #52

## Pull Request Type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation changes

## Checklist

- [x] My code follows the code style of this project.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/designsystem/src/main/java/com/crisiscleanup/core/designsystem/component/Button.kt [hunk 1]
```diff
@@ -282,7 +283,7 @@ fun WorkTypePrimaryAction(
     onClick: () -> Unit = {},
 ) = CrisisCleanupButton(
     // TODO Common dimensions
-    modifier = Modifier.widthIn(100.dp),
+    modifier = Modifier.testTag("workTypePrimaryAction_$text").widthIn(100.dp),
```
Comment (hueachilles) [inline core/designsystem/src/main/java/com/crisiscleanup/core/designsystem/component/Button.kt:286]: Normally the test tags would be constant string identifiers. The text would then be ascertained by accessing the text property. Is that no possible here?

Comment (hueachilles) [reply core/designsystem/src/main/java/com/crisiscleanup/core/designsystem/component/Button.kt:286]: Nvm. I get what you're doing. You want to differentiate between multiple buttons on the same screen. Ignore the comment above.

**Review by hueachilles — APPROVED** [review]
I'll approve since the one change that may be made is inconsequential.

File: feature/authentication/src/main/java/com/crisiscleanup/feature/authentication/AuthenticateScreen.kt [hunk 2]
```diff
@@ -219,7 +221,7 @@ private fun LoginScreen(
                 }
             }
             BusyButton(
-                modifier = fillWidthPadded,
+                modifier = fillWidthPadded.testTag("loginLoginDebugBtn"),
```
[the remark this answers is not in the document]
Comment (deepanchal) [reply feature/authentication/src/main/java/com/crisiscleanup/feature/authentication/AuthenticateScreen.kt:224]: Yes. I agree. I just added a testTag incase we want to test it in future.

**Pull request merged**